.github/workflows/tests.yml

```diff
@@ -27,3 +27,13 @@ jobs:
           python -m pip install --editable .
       - name: "Run tests for ${{ matrix.python-version }}"
         run: python -m pytest
+      - name: "Run mypy checks for ${{ matrix.python-version }}"
+        run: |
+          python -m pip install mypy==1.4.1
```

Comment:
Why is this pinned, and why to such old version?
mypy 1.8.0 seemed to work as far as I could tell

Reply:
Pinning to avoid any issues if they tighten something up later on and a random CI build here breaks (which IIRC I've seen before elsewhere). 1.4.1 because it's the last version with 3.7 support (they removed in https://github.com/python/mypy/commit/2edaf35ec1f91c139dfe1930b1b0d1e1cac59599), which I will note is [only 8 months old](https://github.com/python/mypy/releases/tag/v1.4.1)!